crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -158,6 +158,21 @@ async fn check_gas(
             TransactionKind::Single(SingleTransactionKind::PaySui(t)) => t.amounts.iter().sum(),
             _ => 0,
         };
+
+        // Ensure that a supplied transaction has specified a sufficiently high gas price,
+        // otherwise reject the transaction
+        let reference_gas_price = store
```

Comment:
Unfortunately this doesn't work at epoch boundaries. Specifically, the reference price from the sui system state object isn't guaranteed to be consistent with the current epoch

Reply:
the price shouldn't ever be updated until the change epoch transaction is executed so this should technically work. Although if you'd prefer handling this another way how would you suggest we do this, given we need something to land by end of day.

Reply:
But that transaction isn't guaranteed to be executed as the last transaction, right? The checkpoint executor could execute tx out of order (purely dependency driven)